### DISCUSSION TITLE
Move the source code storage to Codeberg.org

### DISCUSSION THREAD

**Discussion by linux-lukas** [VSCodium/vscodium#968] [General]
Would it be possible to migrate the source code from GitHub to Codeberg (based on Gitea; located in Germany (EU))? 

This would free the project from the general dependency, the fact that GitHub belongs to Microsoft.

**Comment by linux-lukas** [comment]
You could build a mirror from Codeberg to GitHub or from GitHub to Codeberg.

**Comment by paulcarroty** [comment]
CI? Pull requests? Discussions? The import will take a gigantic amount of time.

**Reply by linux-lukas** [reply]
On the one hand, you are of course right that a migration from GitHub to Codeberg would take some time in any case - on the other hand, however, one has to see that the effort would be enormously worthwhile, since Codeberg.org is run by a non-profit, non-governmental organisation (NGO) and it would therefore be difficult to be bought out in the same way as Github. 

As far as I have seen, it is planned that features such as Discussions will also be available in Gitea (on which Codeberg.org is based) in the medium and long term. 

Until then, however, it would certainly not be unwise to consider and take preparatory steps for a migration.

**Comment by dminca** [comment]
Definitely an awesome idea.

On the same note, would be cool to login on Codeberg from the IDE instead of GitHub 😄

**Reply by dminca** [reply]
only bottleneck I see is the CI/CD pipeline, which GitHub Actions is pretty mature and rubust, while Codeberg is still building its way through Woodpecker CI

**Reply by linux-lukas** [reply]
Registration via GitHub or Gitlab is now also possible: https://mastodon.technology/@codeberg/108446852222072224

**Reply by dminca** [reply]
ok, but does the IDE accept the Codeberg token?
